src/attpc_engine/detector/transporter.py

```diff
@@ -0,0 +1,441 @@
+import math
+import numpy as np
+
+from numba import njit, float64, int64
+from numba.types import UniTuple, DictType
+from numba.typed import Dict
+from numba.experimental import jitclass
+from .parameters import Detector_Params, Electronics_Params, Pad_Params, Parameters
+from .beam_pads import BEAM_PADS as b_pads
+
+NUM_TB: int = 512
+STEPS = 10
+
+
+@njit
+def normal_pdf(x: float, mu: float, sigma: float):
+    """
+    Equation of PDF corresponding to a 1D normal distribution.
+
+    Parameters
+    ----------
+    x: float
+        Value to evaluate PDF at.
+    mu: float
+        Mean of distribution.
+    sigma: float
+        Standard deviation of distribution.
+
+    Returns
+    -------
+    float
+        Value of distribution with the input mean and sigma
+        at the input value.
+    """
+    c1: float = 1 / math.sqrt(2 * np.pi) / sigma
+    c2: float = (-1 / 2) * ((x - mu) / sigma) ** 2
+
+    return c1 * math.exp(c2)
+
+
+@njit
+def bivariate_normal_pdf(
+    point: tuple[float, float], mu: tuple[float, float], sigma: float
+):
+    """
+    Equation of PDF corresponding to a 2D normal distribution where sigma is the same
+    for both x and y and there is no correlation between the two variables.
+
+    Parameters
+    ----------
+    point: tuple[float, float]
+        Point to evaluate PDF at.
+    mu: tuple[float, float]
+        Mean of distribution.
+    sigma: float
+        Standard deviation of distribution. Same for both
+        x and y.
+
+    Returns
+    -------
+    float
+        Value of distribution with the input mean and sigma
+        at the input point.
+    """
+    c1: float = 1 / 2 / np.pi / (sigma**2)
+    c2: float = (-1 / 2 / sigma**2) * (
+        ((point[0] - mu[0]) ** 2) + ((point[1] - mu[1]) ** 2)
+    )
+
+    return c1 * math.exp(c2)
+
+
+@njit
+def meshgrid(xarr: np.ndarray, yarr: np.ndarray):
+    """
+    Creates a rectangular (x, y) grid from the input arrays and
+    returns a Nx2 array where each row is a point in the mesh.
+
+    Parameters
+    ----------
+    xarr: np.ndarray
+        1xN array of x values of mesh.
+    yarr: np.ndarray
+        1xN array of y values of mesh.
+
+    Returns
+    -------
+    np.ndarray
+        Nx2 array of points in mesh where the
+        first column is the x-value and the second
+        column is the y-value.
+
+    """
+    grid = np.empty(shape=(xarr.size * yarr.size, 2), dtype=np.float64)
+    idx: int = 0
+
+    for x in xarr:
+        for y in yarr:
+            grid[idx][0] = x
+            grid[idx][1] = y
+            idx += 1
+
+    return grid
+
+
+key_type = int64
+value_type = int64[:]
+
+transport_spec = [
+    ("center", UniTuple(float64, 2)),
+    ("electrons", int64),
+    ("time", float64),
+    ("sigma_t", float64),
+    ("sigma_l", float64),
+    ("pads_hit", DictType(key_type, value_type)),
+]
+
+
+@jitclass(spec=transport_spec)
+class TransportPoint:
```

Comment:
TransportPoint is kinda interesting. I think we could simplify this to a helper class. Instead of taking in and storing an electron count/time (and therefore allocating one TransportPoint per point in the trajectory, 33401 of them) we allocate one Transporter where we then manually call find_pads_hit and pass the electron number and time to that function

Reply:
Mostly cause I'm not sure what the overhead of allocating that many jitclasses is

Reply:
In Spyral all of our jitclasses are allocated more or less once per process. If they compile every time, that could really slow things down